logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -51,12 +51,12 @@ def fetch_or_store(namespaces, key, default_value = nil)
       # BUT. If the value is not present in the `@fast_lookup` the value will be inserted and we assume that we don't
       # have it in the `@metric_store` for structured search so we add it there too.
 
-      value = @fast_lookup.get(namespaces.dup << key)
+      value = @fast_lookup.get((namespaces.dup << key).hash)
```

Comment:
@jakelandis yes I mean the hash itself is 32 bit here it seems and chances are we're safe, but there could be effects that make collisions very likely as a result of JRuby's implementation :)
http://preshing.com/20110504/hash-collision-probabilities/ for math, but note that we likely don't have 32 clean bits in our hashes here :)